### DISCUSSION TITLE
msnoise hourly CC

### DISCUSSION THREAD

**Discussion by sujan1746** [ROBelgium/MSNoise#244] [Q&A] [answered]
I am so sorry to ask you again about hourly cross correlation.

I need to get hourly cross correlation and want to understand the parameters in detail.

I know you told me that the msnoise is not available to get hourly dv/v, and for getting hourly CC, I need to change more than just modifying corr_duration=3600.

So my questions are..
**1. Whether I need to change the data structure SDS to IDDS.**
I changed the data structure SDS to IDDS but because of my lack of skill, I got some problems in process. (I think it would take some time to solve them)
So I wonder whether I have to change the data structure to IDDS or not.

**2. Which parameters I need to change?**
I changed the parameter `keep_all=Y `and` corr_duration=3600`.
In issue #234 , I got the reply that 'No hourly CCF can be obtained using ``corr_duration = 3600` only.'
When I see the configuration parameters, it is hard to find other parameters that I need to change for computation of hourly CC.

I can only see the corr_duration and cc_sampling_rate related to computing hourly CC.
But I am not sure whether I need to change cc_sampling_rate(20.0) or not.
Could you please let me know which parameters that I need to change more?

p.s. Actually I am not sure how to determine the value of cc_sampling_rate and overlap.
How do you get the standard (consideration keys) to determine them?

**3. Is it possible to compute stacking by using hourly CC?**
In issue #234 , it is possible to get hourly CC but impossible to get hourly dvv.
Between the two steps, there are steps of stacking, compute_mwcs, and compute_dtt.
Then from which step cannot be computed hourly?
I want to know this because if I succeed to compute hourly CC, I hope to use both CC and Stacking data.
When I tried to get the dispersion graph, I used stacking data because of format.
So I wonder whether it is able to compute stacking by using hourly CC?

**Comment by ThomasLecocq** [comment, marked as answer]
> I need to get hourly cross correlation and want to understand the parameters in detail.
> 
> I know you told me that the msnoise is not available to get hourly dv/v, and for getting hourly CC, I need to change more than just modifying corr_duration=3600.
> 
> So my questions are.. **1. Whether I need to change the data structure SDS to IDDS.** I changed the data structure SDS to IDDS but because of my lack of skill, I got some problems in process. (I think it would take some time to solve them) So I wonder whether I have to change the data structure to IDDS or not.

No, by default, msnoise will use any structure (or non-structure using ``scan_archive --path ...`` ,any file duration (even 5 minutes files) and *make use of them to compute 1-day long correlations* !

> 
> **2. Which parameters I need to change?** I changed the parameter `keep_all=Y `and` corr_duration=3600`. In issue #234 , I got the reply that 'No hourly CCF can be obtained using ``corr_duration = 3600` only.' When I see the configuration parameters, it is hard to find other parameters that I need to change for computation of hourly CC.

Yes, that's the only two to change to obtain hourly cross-correlation functions (CCFs). In #234 I told you that hourly CCFs are the *only* hourly product you can obtain from msnoise currently. The next steps (STACK, MWCS, DTT, DVV) are *not* built for dealing with hourly data, only the 1-day stacks.

> I can only see the corr_duration and cc_sampling_rate related to computing hourly CC. But I am not sure whether I need to change cc_sampling_rate(20.0) or not. Could you please let me know which parameters that I need to change more?

``cc_sampling_rate`` is the ``Sampling Rate for the CrossCorrelation (in Hz)`` : all traces will be downsampled to this sampling rate *prior* to correlation. This parameter (default to 20 Hz, which is OK for standard 0.1 - 1.0 Hz CC) is of course to be adjusted if you want to use higher frequencies.

> p.s. Actually I am not sure how to determine the value of cc_sampling_rate and overlap. How do you get the standard (consideration keys) to determine them?

for ``overlap`` well that depends on the cases, but e.g. a ``corr_duration=3600`` and ``overlap=0.5`` will output results with a 30 min granularity in the ``keep_all`` H5 files.

> **3. Is it possible to compute stacking by using hourly CC?** In issue #234 , it is possible to get hourly CC but impossible to get hourly dvv. Between the two steps, there are steps of stacking, compute_mwcs, and compute_dtt. Then from which step cannot be computed hourly? I want to know this because if I succeed to compute hourly CC, I hope to use both CC and Stacking data. When I tried to get the dispersion graph, I used stacking data because of format. So I wonder whether it is able to compute stacking by using hourly CC?

No, you'll have to do it yourself. You should make use of the API function ``get_results_all``, apply cleaning/filtering if needed, then stack and save to disk.

**Reply by sujan1746** [reply]
I see.
Then I do not need to care about the data structure anymore (I will use SDS again) and will care about other steps (STACK, MWCS, DTT, DVV) for hourly computation!
And thanks for your comment, I understand cc_sampling_rate and overlap more than before :)

There are some curious things related to corr_duration.

If the keep_all=Y and corr_duration=3600 make hourly CC, I guess it means cross correlation with 3600s=1h.
However, the default corr_duration value is 1800 (30min) and the default of CC is daily.
Could you please let me know what I am missing?

And is there any way to check whether the CC is hourly or daily?

p.s. In documentation, N=1day/corr_duration. What does the N exactly mean?

**Reply by ThomasLecocq** [reply]
> If the keep_all=Y and corr_duration=3600 make hourly CC, I guess it means cross correlation with 3600s=1h.
> However, the default corr_duration value is 1800 (30min) and the default of CC is daily.
> Could you please let me know what I am missing?

I don't understand what you don't get. The ``corr_duration`` is the length of individual windows that will be cross-correlated, saved to the large H5 file in CROSS_CORRELATION folder (if ``keep_all=Y``) and their average to a single MSEED file in STACKS/<filterid>/001_DAYS (if ``keep_days=Y``)

>And is there any way to check whether the CC is hourly or daily?

The CC is always of ``corr_duration``  length. Then the, e.g, 24 hourly CC are stacked to 1 day and save to the STACKS folder.

> p.s. In documentation, N=1day/corr_duration. What does the N exactly mean?

N = Number of correlations. (this equation is valid if ``overlap=0``, of course)

**Reply by sujan1746** [reply]
I understand what you replied and why the `corr_duration `setting should be 3600s for hourly CC because CC is of `corr_duration `length.
I wonder why 30min(1800s) length of `corr_duration `is able to result in daily CC.

To be more specific, CC is of `corr_duration `length.
The default `corr_duration `is 1800s=30min.
The default CC is daily CC.

If the `corr_duration `is the factor that determine whether the CC would be daily or hourly, then if the `corr_duration=1800s`, why is the result CC not 30min CC but 1 day CC?

I know it looks a silly question but want to understand more..

**Comment by ThomasLecocq** [comment]
I can't find a better way to explain than what I just did.
http://msnoise.org/doc/workflow/005_compute_cc.html#computing-the-cross-correlations shows the flow chart of this step.

**Reply by sujan1746** [reply]
I got it. I will try more.
Thank you for helping me to know :)
